examples/tf/example_tensorboard_logger.py

```diff
@@ -1,19 +1,13 @@
 import tensorflow as tf
 
-from garage.misc import logger
+from garage.logger import logger, tabular, TensorBoardOutput
 
-logger.set_tensorboard_dir("data/local/histogram_example")
+logger.add_output(TensorBoardOutput("data/local/histogram_example"))
 N = 400
 for i in range(N):
     sess = tf.Session()
     sess.__enter__()
     k_val = i / float(N)
-    logger.record_histogram_by_type('gamma', key='gamma', alpha=k_val)
-    logger.record_histogram_by_type(
-        'normal', key='normal', mean=5 * k_val, stddev=1.0)
-    logger.record_histogram_by_type('poisson', key='poisson', lam=k_val)
-    logger.record_histogram_by_type(
-        'uniform', key='uniform', maxval=k_val * 10)
-    logger.record_tabular("app", k_val)
-    logger.record_histogram("gass", k_val)
-    logger.dump_tensorboard(step=i)
+    tabular.record("app", k_val)
+    logger.log(("gass", k_val), record='histogram')
```

Comment:
The `record` parameter is specific to `TensorBoardOutput` but not in base logger interface. I think we should not pass ad hoc configuration to `Output` when calling `logger.log()` - all output options should be configured through only `output_instance` but not logger interface.

In this particular case, this `record` parameter is instructing logger to log as histogram. What about using a `Histogram` class to wrap up data, just like using `Tabular` to log tabular.

Also, instantiating helper wrapper every time can be pretty annoying and cumbersome. I also suggest add sugar methods like `logger.scalar()`, `.histogram()`, `.text()` and `.tabular()` to logger.

Reply:
`tf.summary` as a ref for my last point: https://www.tensorflow.org/api_docs/python/tf/summary/histogram

Reply:
i agree tho we can keep that as an issue for future enhancement. let's focus on the API and design fundamentals in this PR.

Reply:
@naeioi I like this idea! Definitely something that can be added later